### PR TITLE
Remap source path prefix

### LINKED ISSUE #71: Compiled binaries depend on the crate source code
When I build my [rust project](https://github.com/Xe/withinbot) as a Docker image, I get a ~200 MB image. This seems to be because every single one of the crates it depends on gets included into the resulting binary. Here is how I am building a docker image:

- https://github.com/Xe/withinbot/blob/master/docker.nix
- https://github.com/Xe/withinbot/blob/master/default.nix

How do I remove references to the crates from the compiled binary? The paths to the crates seem to live in the Rust panicinfo crud.

### PR DESCRIPTION
**Pull request by nmattia**
This adds the option `remapPathPrefix` which, when set to `true`, reduces the closure size by avoiding references to the source files in the nix store.

Closes #71 

Was #74 

CC @andrew-d

try this out: `niv update naersk -b nm-strip-refs`

### CODE REVIEW

File: build.nix [hunk 1]
```diff
@@ -220,6 +226,25 @@ let
...
+      # Remove the source path(s) in Rust
+      if [ -n "$RUSTFLAGS" ]; then
+        RUSTFLAGS="$RUSTFLAGS --remap-path-prefix $crate_sources=/sources"
+        log "RUSTFLAGS (updated): $RUSTFLAGS"
+      elif [ -n "$CARGO_BUILD_RUSTFLAGS" ]; then
+        CARGO_BUILD_RUSTFLAGS="$CARGO_BUILD_RUSTFLAGS --remap-path-prefix $crate_sources=/sources"
+        log "CARGO_BUILD_RUSTFLAGS (updated): $CARGO_BUILD_RUSTFLAGS"
+      else
+        export CARGO_BUILD_RUSTFLAGS="--remap-path-prefix $crate_sources=/sources"
+        log "CARGO_BUILD_RUSTFLAGS (updated): $CARGO_BUILD_RUSTFLAGS"
```
Comment (basvandijk) [inline build.nix]:
```suggestion
        log "CARGO_BUILD_RUSTFLAGS: $CARGO_BUILD_RUSTFLAGS"
```

File: build.nix [hunk 2]
```diff
@@ -220,6 +226,25 @@ let
       log "cargo_test_options: $cargo_test_options"
       log "cargo_bins_jq_filter: $cargo_bins_jq_filter"
       log "cargo_build_output_json (created): $cargo_build_output_json"
+      log "crate_sources: $crate_sources"
+      log "RUSTFLAGS: $RUSTFLAGS"
+      log "CARGO_BUILD_RUSTFLAGS: $CARGO_BUILD_RUSTFLAGS"
+
+      ${lib.optionalString remapPathPrefix ''
+
+      # Remove the source path(s) in Rust
```
Comment (basvandijk) [inline build.nix]: Maybe indent this block so that you can clearly see where the block begins and ends. But it's style so your call.

Comment (nmattia) [reply build.nix]: let's ask `nixpkgs-fmt`

File: build.nix (hunk 1 above)
Comment (nmattia) [reply build.nix]: what do you mean? I'm printing a non-updated version above, here I want to point out that it's been updated.

File: build.nix (hunk 2 above)
Comment (nmattia) [reply build.nix]: `nixpkgs-fmt` is happy, though it does complain about `symlinkJoinPassViaFile` _cough cough_

**Pull request merged**